internal/expr/query.go

```diff
@@ -186,6 +204,13 @@ func (qe *QueryExpr) ScanArgs(columns []string, outputArgs []any) (scanArgs []an
 		for _, mi := range mapSetInfos {
 			mi.mapVal.SetMapIndex(mi.keyVal, mi.scanVal)
 		}
+		for _, si := range scanInfos {
+			if !si.pointerScanVal.IsNil() {
+				si.fieldVal.Set(si.pointerScanVal.Elem())
+			} else {
+				si.fieldVal.Set(reflect.Zero(si.fieldVal.Type()))
```

Comment:
This will change given the above, but FWIW the common logic may be pulled out of the branch and done after it (si.fieldVal.Set).

Reply:
Done.